src/grid/onedgrid.py

```diff
@@ -327,17 +327,15 @@ def RectangleRuleSineEndPoints(npoints):
 
     weights = np.zeros(npoints)
 
-    index_m = np.arange(npoints) + 1
+    m = np.arange(npoints) + 1
 
     for i in range(0, npoints):
-        elements = np.zeros(npoints)
-        elements = np.sin(index_m * np.pi * points[i])
-        elements *= (1 - np.cos(index_m * np.pi)) / (index_m * np.pi)
-
-        weights[i] = (2 / (npoints + 1)) * np.sum(elements)
+        weights[i] = np.sum(
+            np.sin(m * np.pi * points[i]) * (1 - np.cos(m * np.pi)) / (m * np.pi)
+        )
```

Comment:
Here is still a ``for`` loop. Could you please vectorize the way weights are computed so that all weights are computed at once.